src/test/java/com/driving/school/service/CrudServiceTest.java

```diff
@@ -33,34 +42,283 @@ class CrudServiceTest {
     private StubCrudService service;
 
     @Test
-    @DisplayName("Can getAll entities")
-    void canGetAll() {
+    @DisplayName("getAll(): gets a list entities")
+    void whenGetAll_thenOK() {
         //given
-        List<String> dbContent = List.of("a", "bc", "def");
-        List<StubResponseDto> responseMappings = dbContent.stream()
+        List<String> dbContent = createDbContent();
+        List<StubResponseDto> responseMappings = setUpMock_mapper_toResponseDto(dbContent);
+        //and
+        setUpMock_repo_findAll(dbContent);
+
+        //when
+        List<ResponseDto> actual = service.getAll();
+
+        //then
+        verify(repo).findAll();
+        assertAllDtosAreCorrect(responseMappings, actual);
+    }
+
+    @Test
+    @DisplayName("getAll(): on an empty DB, returns empty list")
+    void whenGetAll_thenEmptyList() {
+        //given
+        setUpMock_repo_findAll(new ArrayList<>());
+
+        //when
+        List<ResponseDto> actual = service.getAll();
+
+        //then
+        verify(repo).findAll();
+        assertNoDtoWasReturned(actual);
+    }
+
+    @Test
+    @DisplayName("getById(): gets an entity")
+    void whenGetById_thenOK() {
+        //given
+        String foundEntity = setUpMock_repo_findById();
+        //and
+        StubResponseDto expectedResponseMapping = setUpMock_mapper_toResponseDto(foundEntity);
+
+        //when
+        ResponseDto actual = service.getById(1L);
+
+        //then
+        verify(repo).findById(anyLong());
+        assertCorrectDtoIsReturned(expectedResponseMapping, actual);
+    }
+
+    @Test
+    @DisplayName("patchById(): updates an existing entity")
+    void whenPatchById_thenOK() {
+        //given
+        setUpMock_repo_findById();
+        //and
+        String patchedEntity = setUpMock_repo_save();
+        //and
+        StubResponseDto expectedResponseMapping = setUpMock_mapper_toResponseDto(patchedEntity);
+        //and
+        //anonymous empty implementation (notice the {})
+        UpdateDto requestData = new UpdateDto() {};
+
+        //when
+        ResponseDto actual = service.patchById(1L, requestData);
+
+        //then
+        verify(repo).findById(anyLong());
+        assertCorrectDtoIsReturned(expectedResponseMapping, actual);
+    }
+
+    @Test
+    @DisplayName("create(): creates and saves an entity")
```

Comment:
Nie wiem czy to wynika z ubogiego nazewnictwa ale tak naprawdę to nie tworzymy w metodzie `service.save()`,  żadnego obiektu, a raczej zapisujemy, mapujemy i zwracamy ResponseDto

```suggestion
@DisplayName("create(): Saves an object as entity in db")
```
- to może być błędna sugestia, ale proszę spróbuj objaśnić

Reply:
Jak najbardziej tworzymy (w metodzie `service.create()`, zakładam że o tę Ci chodziło i `service.save()` to pomyłka). Metoda `mapper.toModel()` tworzy nowy obiekt (encję). Jeśli nie wierzysz to sprawdź sobie implementacje tych metod, a konkretnie wybierz dowolną implementację mappera, np. `InstructorMapperImpl` i zobacz metodę `createEntity()`, wywoływaną właśnie przez `mapper.toModel()`.

Niemniej Twoja nazwa może być ;)

Reply:
Tak, masz rację `service.create()`. Jeśli nazwa nie musi być spójna z tym co się dzieje dokładnie, a raczej żeby lepiej się czytało co te testy robią. to niech zostanie, a Jak nie to cos w stylu `Create Dto and Save object as entity in db`. ( możne ta forma bezosobowa nawet lepsza jest)